app/controllers/mileages_controller.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class MileagesController < ApplicationController
+  def index
+    @mileages = current_user.mileages.order(:car_id, meter_status: :asc).page params[:page]
+  end
+
+  def show
+    @mileage = current_user.mileages.find(params[:id])
```

Comment:
You don't need using 'current_user' to fetch the wanted Mileage. Please use ``` Mileage.find(params[:id]) ```

I know this solution was earlier used in other 'controllers' but I didn't notice it before.

Reply:
Do we need restrict user mileages to be visible only for current user?

Reply:
@nikos83  It's a 'show' action which returns only ONE specific record defined by his ID.

Reply:
I know but at the moment on the top url you can change id

Reply:
@nikos83 And what's the problem?

Reply:
I think that @nikos83 is right. User should have access only for his entries. So, we just can add rescue for this (at this moment we have query error:
`ActiveRecord::RecordNotFound in MileagesController#show
Couldn't find Mileage with 'id'=7 [WHERE "mileages"."deleted_at" IS NULL AND "cars"."deleted_at" IS NULL AND "cars"."user_id" = $1]`

Reply:
ok :)